src/App.tsx

```diff
@@ -1,204 +1,89 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+export enum ErrorType {
+  TodosLoad = 'Unable to load todos',
+  EmptyTitle = 'Title should not be empty',
+  UnableToAddTodo = 'Unable to add a todo',
+  UnableToDeleteTodo = 'Unable to delete a todo',
+  UnableToUpdateTodo = 'Unable to update a todo',
+}
+
+export enum Filter {
+  all = 'All',
+  active = 'Active',
+  completed = 'Completed',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isTodoEditing, setIsTodoEditing] = useState(false);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [currentError, setCurrentError] = useState<ErrorType | ''>('');
+  const [selectedFilter, setSelectedFilter] = useState(Filter.all);
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setCurrentError(ErrorType.TodosLoad);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (!currentError) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setCurrentError('');
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [currentError]);
+
+  const activeTodos: number = [...todos].filter(
```

Comment:
```suggestion
  const activeTodos: number = todos.filter(
```
filter does not mutate an array